docs/DeveloperGuide.md

```diff
@@ -1,38 +1,462 @@
+
+<br/>
+
+**Section 1: Parse and Validation of Information**
+
+The first section is facilitated by the following classes:
+
+- `ParseAdd`: Contains common methods used by `ParseAddClient` and `ParseAddProperty`.
+- `ParseAddClient`: Extracts and validates client information from `commandDescription`(User Input).
+- `ParseAddProperty`: Extracts and validates property information from `commandDescription`(User Input).
+
+The following is a simple class diagram of the three classes:
+<p align="center">
+
+![](diagrams/ParseAddRelatedClassesDiagram.jpg)
+
+</p>
+
+<p align="center">
+Parse Add Related Classes Diagram
+</p>
+
+As shown above, both `ParseAddClient` and `ParseAddProperty` classes have a similar core method called `parseCommand()` which is responsible for client or property detail extraction and validation. The rest of the methods in both classes are sub-methods of the `parseCommand()` method.
+
+Also, most of the sub-methods are used to perform validations on the extracted details. These methods are implemented via regex pattern checker.
+
+- Client:
+    - `checkForValidSingaporeContactNumber(String)`
+    - `checkForValidEmail(String)`
+    - `checkForBudgetNumberFormat(String)`
+- Property:
+    - `checkForValidSingaporeAddress(String)`
+    - `checkForValidSingaporeLandedPropertyAddress(String)`
+    - `checkForValidSingaporeBuildingAddress(String)`
+    - `checkForPriceNumberFormat(String)`
+- Common:
+    - `checkForEmptyDetails(String)`: Checks for any missing essential details, non-essential detail such as optional email can be empty.
+
+Note: Since the target user is a property manager working in Singapore, some validations are tailored to Singapore context.
+
+<br/>
+
+**Section 2: Addition of client or property to client list or property list**
+
+The second section is facilitated by the following classes: 
+- `CommandAdd`: Abstract superclass of `CommandAddClient` and `CommandAddProperty` classes.
+- `CommandAddClient`: Creates a `Client` object and add it to the `clientList`.
+- `CommandAddProerty`: Creates a `Property` object and add it to the `propertyList`.
+
+The following is a simple class diagram of the three classes:
+
+<p align="center">
+
+![Command Add Related Classes Diagram](diagrams/CommandAddRelatedClassesDiagram.jpg)
+
+</p>
+
+<p align="center">
+Command Add Related Classes Diagram
+</p>
+
+
+As shown above, both `CommandAddClient` and `CommandAddProperty` classes have a similar core method called `execute(...)` which is responsible for the new client or property addition into their respective lists.
+
+<br/>
+
+**Example Scenario**
+
+Given below is an example scenario on how add client/property behaves at each step.
+
+
+- **Step 1**: The user executes ```add -client n/NAME c/CONTACT_NUMBER e/EMAIL b/BUDGET_MONTH``` or ```add -property n/NAME a/ADDRESS p/PRICE t/TYPE```. Depending on `add -client` or `add -property` specified, a `Parser` object of type `ParseAddClient` or `ParseAddProperty` is created.
```

Comment:
Consider giving high level and more easily readable steps. Since methods are not individually explained, perhaps there is no need to quote specific methods.